### PR TITLE
[Narwhal] make tx cannons configurable from cli

### PR DESCRIPTION
**Pull request by niklaslong**
This PR introduces a basic `clap`-based CLI to make the `simple_nodes` example more configurable for devnet purposes. The tx cannons can now be enabled/disabled and configured.

### CODE REVIEW

File: node/narwhal/Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,9 @@ version = "1.0"
 [dependencies.bytes]
 version = "1"
 
+[dependencies.clap]
```
Comment (howardwu) [inline node/narwhal/Cargo.toml]: dev-dep

File: node/narwhal/examples/README.md [hunk 2]
```diff
@@ -19,35 +19,47 @@ cargo run --release -- example monitor
 
 ## Development
 
+```
+Usage: simple_node [OPTIONS] --mode <MODE> --node-id <ID> --num-nodes <N>
+
+Options:
+      --mode <MODE>                   The mode to run the node in [possible values: narwhal, bft]
```
Comment (ljedrz) [inline node/narwhal/examples/README.md]: these docs could use a bit of information on what the difference is between these modes

File: node/narwhal/examples/README.md [hunk 3]
```diff
@@ -57,3 +69,8 @@ The file must contain the peer node IDs, IP addresses and ports, in the followin
 ```
 
 If this parameter is not present, all nodes are run on localhost.
+
+In addition, `--fire-cannons` will enable the transaction and solution cannons for each node.
+If enabled, the interval in milliseconds can optionally be passed in as an argument.
+
+
```
Comment (ljedrz) [inline node/narwhal/examples/README.md]:
```suggestion
```

File: node/narwhal/examples/simple_node.rs [hunk 4]
```diff
@@ -398,6 +399,32 @@ async fn start_server(bft: Option<BFT<CurrentNetwork>>, primary: Primary<Current
 
 /**************************************************************************************************/
 
+#[derive(Debug, Clone, ValueEnum)]
+enum Mode {
```
Comment (ljedrz) [inline node/narwhal/examples/simple_node.rs:404]: this `enum` and its variants could use some docs

**Review by ljedrz — APPROVED** [review]
LGTM, left some suggestions

File: node/narwhal/examples/simple_node.rs [hunk 5]
```diff
@@ -398,6 +399,35 @@ async fn start_server(bft: Option<BFT<CurrentNetwork>>, primary: Primary<Current
...
+
+/// A simple CLI for the node.
+#[derive(Parser, Debug)]
+struct Args {
+    /// The mode to run the node in.
+    #[arg(long)]
+    mode: Mode,
+    /// The ID of the node.
+    #[arg(long, value_name = "ID")]
+    node_id: u16,
```
Comment (howardwu) [inline node/narwhal/examples/simple_node.rs]: nit: I'd prefer this be `id`

File: node/narwhal/examples/simple_node.rs [hunk 6]
```diff
@@ -398,6 +399,35 @@ async fn start_server(bft: Option<BFT<CurrentNetwork>>, primary: Primary<Current
...
+    /// The number of nodes in the network.
+    #[arg(long, value_name = "N")]
+    num_nodes: u16,
+    /// If set, the path to the file containing the committee configuration.
+    #[arg(long, value_name = "PATH")]
+    config: Option<PathBuf>,
+    /// Enables the tx and solution cannons, and optionally the interval in ms to run them on.
+    #[arg(long, value_name = "INTERVAL_MS")]
+    fire_cannons: Option<Option<u64>>,
```
Comment (howardwu) [inline node/narwhal/examples/simple_node.rs]: nit: I'd prefer this be broken up into `fire-solutions` and `fire-transactions`. We can have a catch all that fires all with `fire-transmissions` or `fire-all` (you're preference between those two)

File: node/narwhal/examples/simple_node.rs (hunk 5 above)
Comment (niklaslong) [reply node/narwhal/examples/simple_node.rs]: I renamed this in the CLI but didn't touch the usages of `node_id` that predate the PR, happy to adjust these as well, if that is preferred.

File: node/narwhal/examples/simple_node.rs (hunk 6 above)
Comment (niklaslong) [reply node/narwhal/examples/simple_node.rs]: Done, went with `fire-transmissions` to be consistent with our naming for the algebraic type.

**Review by howardwu — APPROVED** [review]

**Pull request merged**